### ISSUE TITLE
[Redesigned Dashboard] | Submission List, editorial activity decline review requests icon is throwing accessibility errors.

### ISSUE THREAD

**Issue by Tribunal33** [pkp/pkp-lib#11162]
## Issue Description
Storybook threw some violations for decline indicator for editorial activity section
https://main--6555d3db80418bb1681b8b17.chromatic.com/?path=/story/components-reviewactivityindicatorpopup--declined
// What is the nature of the issue? How does it impact users?

![Image](https://github.com/user-attachments/assets/12b19f8c-3ec5-4c3e-b229-156231860f3f)

## How to reproduce / Current behavior
1. Have a submission in the review state with reviewer declining request to review.
2. Go to dashboard and observe the icon

## What application are you using?
 OJS, OMP or OPS version 3.5.0 RC2

## Stack used 
- Device: 
- OS: Windows 10
- Browser: Chrome
- Version: latest

## Testing method
- Method: (e.g. Manual; Keyboard)
- Automated test: (e.g. tool name)

## Action
// Describe action that can be taken to remediate the issue.
Storybook provides links to potential fixes.  Given in the additional information section. 

## Additional information
// Please add any screenshots, code snippets, or other information we can use to investigate this issue.
https://dequeuniversity.com/rules/axe/4.10/button-name?application=axeAPI

**Comment by jardakotesovec** [comment]
Thanks, this means that these buttons don't have anything to say for screenreader:

<img width="165" alt="Image" src="https://github.com/user-attachments/assets/53d52f5c-1759-41f2-b04d-6c660c1bc2c9" />

@israelcefrin I would go with following, which I think would give same amount of the information that user without SR can gather from the icon/number if he is familiar with these indication.  Or do you think it should be more brief?
aria-label: Display review details.
aria-description: Review request has been shared with reviewer. Awaiting response in 9 days on 03.04.2025

Would be basically reusing the description from the modal:

<img width="490" alt="Image" src="https://github.com/user-attachments/assets/ac293cb1-0b66-454d-a6c6-34c4ad83f104" />

**Comment by Devika008** [comment]
Hi @jardakotesovec ,

For screen readers, I suggest having them read only **"Awaiting response from the reviewer in 9 days"** when announcing the icon. Thats what normally users would read and understand too. Users can then be **prompted to click the icon** to open the modal and access the **extended information**. Reading the full text upfront could lead to unnecessary **repetition and cognitive overload**.

**Comment by israelcefrin** [comment]
Hello @jardakotesovec 

The use of `aria` attributes are technically correct for both. `aria-label` is meant to be used for concise information while `aria-description` is meant to convey longer text. However, checking the whole component, you already have the information that is announced by the latter element stated in the modal that will be open. I think this is @Devika008 's main concern. 

In such case , it is recommended to use `aria-describedby` attribute. You just need to list the ID of the text element which is contained in the hidden modal instead of repeating the same string in different places.

> The elements linked via `aria-describedby` don't need to be visible. It is possible to reference an element even if that element is hidden. For example, a form control can have a description that is hidden by default and revealed on request using a disclosure widget like a "more information" icon. The sighted users click on the icon to view the description, while assistive technology users can immediately access it as the description is referenced from that form control with `aria-describedby`.

Source: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-describedby

I think we have the same approach for this element of interaction. To access further information, sighted users will activate the button while screen reader users will have immediate access to the information that is currently hidden. 

My only concern is that we have two actionable buttons that are also hidden in the modal and I can't see (yet) a way to announce them to non-sighted users unless you open the modal.

**Comment by jardakotesovec** [comment]
@Devika008 Whenever possible we should aim to avoid creating new translations just for the SR. Especially in these cases where there is lots of variances. I think that would be quite a maintenance burden.

So we kind of have to pick and choose either title or the description from the modal, which both describes whats the review status in, just with different level of details.

@israelcefrin Referencing ids between multiple components can be bit tricky for maintenance. So wondering does that actually make any difference to how the SR reads out the information if I use description vs aria-descripedBy?

I intentionally suggested title to be 'Display review details.' Which is action description, rather than icon description to make clear there is action to do.

**Comment by israelcefrin** [comment]
Hello @jardakotesovec 

Technically there are not differences on how they announce the text. However, if you mix languages, e.g.: include a term that should be announced in a second language , `aria-description` may not work properly since it is an attribute and it is not using an existing DOM element that can be parsed by different SR voices. It is the same situation when you can not pause or  go back word by word in a ALT attribute text. 

I would need to test it (using JAWS and VoiceOver) to ensure that it is the same case.

**Comment by jardakotesovec** [comment]
@israelcefrin I see, thats good to know, thanks.

**Comment by jardakotesovec** [comment]
Tried some options today to get the complexity and convinience balance right. Its bit tricky, as in some case we do say the number of days in the title of popup and sometimes not. 

ui-library: https://github.com/pkp/ui-library/pull/568

Here is example for that Awaiting Response example from above, this is what the SR reads (implemented via spans with sr-only class): **View More details. Awaiting Response from the reviewer collapsed**.

Individual indicators can be checked out here - https://6555d3db80418bb1681b8b17-orxqmaayxf.chromatic.com/?path=/story/components-reviewactivityindicatorpopup--response-overdue

@israelcefrin Let me know if you have any concerns with this solution

**Comment by israelcefrin** [comment]
Hello @jardakotesovec 

Thanks for this solution. TBH, I'd rather using sr-only classes than `aria` attributes since the latter is sometimes tricky to use with nested elements structure.
Checking the parsed HTML, my only suggestion would be inverting the text order that is set. I mean, adjusting the HTML to be like this:

```
<button class="text-sm text-gray-900 inline-flex items-center gap-x-1 font-semibold leading-6" id="headlessui-popover-button-v-0" type="button" aria-expanded="false" data-headlessui-state="">
<span class="relative inline-flex items-center justify-center">
<svg height="24" width="24">
<circle stroke="#BBBBBB" fill="transparent" stroke-width="1" r="11.5" cx="12" cy="12"></circle><circle class="stroke-attention" fill="transparent" stroke-dasharray="0 64.527599106404 -11.513223077076228 82.19905778284658" stroke-width="1.5" r="11.25" cx="12" cy="12"></circle><!----></svg>

<span class="sr-only">Review Request overdue by 3 days</span>
<span class="sr-only">View more details</span>

<span aria-hidden="true" class="text-default absolute text-sm-normal">-3</span>
<!----></span>
</button>
```

Using it that way will ensure that the most important information in the icon is announced first ( how long is the overdue date). While the  providing the same information hierarchy to sighted users, i.e.: the number "3" in a circle with a visual stroke in a different colour.

**Comment by jardakotesovec** [comment]
@israelcefrin Ok, swapped the order.

@Tribunal33 Changes are now available in 3.5 and main.

**Comment by Tribunal33** [comment]
This passes.  Storybook is no longer reporting accessibility violations.  This also fixed several other components with similar accessibility issues.  No specific tests were created for this as of yet. Most likely addressed in an accessibilty document space for the future but for now observing Storybook changes is best tests.
